api-server/src/server/middlewares/version-handler.js

```diff
@@ -0,0 +1,17 @@
+const { version } = require('../../../package.json');
+
+// TODO: I have no idea what I have done in the `middleware.json` file, but it might be working.
```

Comment:
Unless the status is returned to the client (that is, intercept the version, and stop processing further handlers), there is a chance the status will be over-written by the next handler that comes along.

Currently, You are getting away with the below because the handler effectively hooking into the last lifecycle event.

```js
res.status(418);
return next();
```